apps/filebrowser/src/filebrowser/api.py

```diff
@@ -394,81 +396,42 @@ def upload_complete(request):
 
 @api_error_handler
 def upload_file(request):
-  """
-  A wrapper around the actual upload view function to clean up the temporary file afterwards if it fails.
-
-  Returns JSON.
-  """
-  pass
-  # response = {}
-
-  # try:
-  #   response = _upload_file(request)
-  # except Exception as e:
-  #   LOG.exception('Upload operation failed.')
-
-  #   file = request.FILES.get('file')
-  #   if file and hasattr(file, 'remove'):  # TODO: Call from proxyFS -- Check feasibility of this old comment
-  #     file.remove()
-
-  #   return HttpResponse(str(e).split('\n', 1)[0], status=500)  # TODO: Check error message and status code
-
-  # return JsonResponse(response)
+  # Read request body first to prevent RawPostDataException later on which occurs when trying to access body after it has already been read
+  body_data_bytes = string_io(request.body)
 
-
-def _upload_file(request):
-  """
-  Handles file uploaded by HDFSfileUploadHandler.
-
-  The uploaded file is stored in HDFS at its destination with a .tmp suffix.
-  We just need to rename it to the destination path.
-  """
   uploaded_file = request.FILES['file']
-  dest_path = request.GET.get('dest')
-  response = {}
-
-  if MAX_FILE_SIZE_UPLOAD_LIMIT.get() >= 0 and uploaded_file.size > MAX_FILE_SIZE_UPLOAD_LIMIT.get():
-    return HttpResponse(f'File exceeds maximum allowed size of {MAX_FILE_SIZE_UPLOAD_LIMIT.get()} bytes.', status=500)
+  dest_path = request.POST.get('destination_path')
 
-  # Use form for now to triger the upload handler process by Django.
-  # Might be a better solution now to try directly using handler in request.fs.upload() for all FS.
-  # form = UploadAPIFileForm(request.POST, request.FILES)
+  # Check if the file type is restricted
+  _, file_type = os.path.splitext(uploaded_file.name)
+  if RESTRICT_FILE_EXTENSIONS.get() and file_type.lower() in [ext.lower() for ext in RESTRICT_FILE_EXTENSIONS.get()]:
+    return HttpResponse(f'File type "{file_type}" is not allowed. Please choose a file with a different extetypension.', status=400)
 
-  if request.META.get('upload_failed'):
-    raise Exception(request.META.get('upload_failed'))  # TODO: Check error message and status code
+  # Check if the file size exceeds the maximum allowed size
+  max_size = MAX_FILE_SIZE_UPLOAD_LIMIT.get()
```

Comment:
Is this the same for all file systems?

Reply:
Yes, this is a hard limit which affects all configured filesystems in Hue.